demisto_sdk/commands/split/jsonsplitter.py

```diff
@@ -124,4 +146,69 @@ def create_module(self):
 
             module_file_path = os.path.join(self.module_dir, file_name)
 
-        write_dict(module_file_path, data=self.module_json_data, indent=4)
+        write_dict(module_file_path, data=self.json_data, indent=DEFAULT_JSON_INDENT)
+
+    def get_auto_output_path(self) -> Tuple[Path, str, str]:
+        """
+        Obtains the output path automatically according to the List name
+        if autocreate_dir == true, otherwise returns the output as is
+        """
+        suffix_by_type = {
+            TypeListData.TEXT: ".txt",
+            TypeListData.HTML: ".html",
+            TypeListData.CSS: ".css",
+            TypeListData.MD: ".md",
+            TypeListData.JSON: ".json",
+        }
+
+        suffix = suffix_by_type.get(self.json_data["type"], ".txt")
+
+        file_name = pascal_case(self.json_data["name"]) + ".json"
+        file_data_name = file_name[: -len(".json")] + "_data" + suffix
+
+        if self.autocreate_dir:
+            pack_name = get_pack_name(self.input)
+
+            if not pack_name:
+                return Path(self.input).parent, file_name, file_data_name
+
+            (lists_dir := Path(PACKS_DIR) / pack_name / LISTS_DIR).mkdir(exist_ok=True)
+
+            # create the specific list dir under the LISTS_DIR if it does not exist
+            # the dir name determine by the file name without the suffix
+            (list_name_dir := lists_dir / Path(file_name).stem).mkdir(exist_ok=True)
+
+            return list_name_dir, file_name, file_data_name
+
+        if not (output := Path(self.output)).is_dir():
+            output = output.parent
+
+        return output, file_name, file_data_name
+
+    def write_file_data(self, list_name_dir: Path, file_data_name: str):
+        if file_data_name.endswith("json"):
+            try:
+                JsonFile.write_file(
+                    json.loads(self.json_data["data"]),
+                    (list_name_dir / file_data_name),
+                    indent=DEFAULT_JSON_INDENT,
+                )
+            except json.decoder.JSONDecodeError as e:
+                raise Exception(
+                    f"Could not parse data of the list {self.json_data['name']}.\n"
```

Comment:
I'd split the reading from writing. The `try` is on the writing but excepts an error in parsing (reading), which is confusing.